gdcdatamodel/validators/link_validator_parser.py

```diff
@@ -0,0 +1,185 @@
+from dictionaryutils import dictionary as gdcdictionary
+from collections import deque
+
+
+class SubgroupParsingNode(object):
+    def __init__(self, parent, schema, code=None):
```

Comment:
in general a lot of the properties on the `SubgroupParsingNode` get initialized elsewhere which makes the logic more difficult to follow. I think this would be easier to reason about if more of the initialization was here in the __init__ rather than computed in functions elsewhere